### PR TITLE
Feature/add new post comment like

### PR DESCRIPTION
**Pull request by williamrolando88**
Hi, in this PR I:
- Created dedicated controllers for `comments` and `likes`
- Create a method to create a new post
- Create a method to add new comment
- Create a method to add new like
- Fix comments and likes values not showing
- Fix comments and likes values not updating

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(params.require(:comment).permit(:text))
+    @comment.author_id = @user.id
+    @comment.post_id = @post.id
```
Comment (rukundoeric) [inline app/controllers/comments_controller.rb:7]: - [ ] <b>REQUIRED:</b> Kindly use `current_user` as it is required by the project. In that case, the code would be something like this: 

```ruby
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    if @comment.save
+      flash[:notice] = "Comment created!"
+      redirect_to user_post_path(@user, @post)
+    else
+      flash[:notice] = "Comment was not created."
+    end
+
+    @comment.comments_counter
```
Comment (rukundoeric) [inline app/controllers/comments_controller.rb:16]: - [ ] <b>REQUIRED:</b> If you put the update comment counter here, even if the comment fails to be saved, the count will still update which is not correct. Kindly fix it.

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class LikesController < ApplicationController
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @like = Like.new(author: @user, post: @post)
```
Comment (rukundoeric) [inline app/controllers/likes_controller.rb:5]: - [ ] <b>REQUIRED:</b> Also here, Kindly use `current_user`

```ruby
  @post = Post.find(params[:post_id])
  @like = @current_user.likes.new(post_id: @post.id)
```

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    if @like.save
+      redirect_to user_post_path(@user, @post)
+    else
+      flash[:notice] = "Error!"
+    end
+
+    @like.likes_counter
```
Comment (rukundoeric) [inline app/controllers/likes_controller.rb:13]: - [ ] <b>REQUIRED:</b>Also here, If you put the update likes counter here, even if the likes fail to be saved, the count will still update which is not correct. Kindly fix it.

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -5,7 +5,28 @@ def index
...
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author_id = @user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```
Comment (rukundoeric) [inline app/controllers/posts_controller.rb:23]: - [ ] <b>REQUIRED:</b> The same goes here, Please use `current_user` as requested by the project. 

```rb
@post = @current_user.posts.new(post_params)
```

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @williamrolando88, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/likes_controller.rb (hunk 4 above)
Comment (williamrolando88) [reply app/controllers/likes_controller.rb:13]: Thank you for the observation

File: app/controllers/likes_controller.rb (hunk 3 above)
Comment (williamrolando88) [reply app/controllers/likes_controller.rb:5]: Thank you for the observation

**Review by hiromataba — APPROVED** [review]
# STATUS: APPROVED :tada: :tada: :tada: 
Hi @williamrolando88        :wave: :wave: 

appr

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**